lib/Alchemy/Phrasea/WorkerManager/Worker/MainQueueWorker.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Alchemy\Phrasea\WorkerManager\Worker;
+
+use Alchemy\Phrasea\Model\Repositories\WorkerJobRepository;
+use Alchemy\Phrasea\WorkerManager\Queue\MessagePublisher;
+
+class MainQueueWorker implements WorkerInterface
+{
+    private $messagePublisher;
+
+    private $repoWorkerJob;
+
+    public function __construct(
+        MessagePublisher $messagePublisher,
+        WorkerJobRepository $repoWorkerJob
+    )
+    {
+        $this->messagePublisher = $messagePublisher;
+        $this->repoWorkerJob    = $repoWorkerJob;
+    }
+
+    public function process(array $payload)
+    {
+
+        // if needed do treatement here depending on the type
+        $queue = null;
+        $messageType = '';
+
+        switch ($payload['type']) {
```

Comment:
we could be more generic by avoiding "switch" : the Q name could be directly translated from the payload type.
That way we don't have to touch the "mainQ" code when a new type is added.
To see / call w. Aina :  
- test if a Q exists to reject bad type ? should not happen... 
- create the Q automatically from the type ?